controllers/fenceagentsremediation_controller_test.go

```diff
@@ -149,7 +149,14 @@ var _ = Describe("FAR Controller", func() {
 			DeferCleanup(k8sClient.Delete, context.Background(), node)
 
 			Expect(k8sClient.Create(context.Background(), underTestFAR)).To(Succeed())
-			DeferCleanup(cleanupFar(), context.Background(), underTestFAR)
+			DeferCleanup(func() {
```

Comment:
> have one cleanup method that runs after each test instead of spreading the cleanup around

Sounds nice and I like to combine logic to simplify a process. ATM I like that for every resource we create we add a cleaning in the next line. Combining them into one place would "break" the `BeforeEach` and `JustBeforeEach` structure. Therefore, I am leaning toward keeping it as it is and maybe change it in the future,

>Also not sure node cleanup is needed, I vaguely remember some issue with that.

In the past, I had some issues when I tried to create a node/resource where the resource version was identical. This could be avoided when the node/resource is deleted and recreated again. Therefore, I think it is needed and some delay between tests is needed so there won't be any race (trying to create a resource which is currently deleting). This could also be avoided if we delete we zero grace period.

Reply:
>Therefore, I am leaning toward keeping it as it is and maybe change it in the future

Fair enough , I consider this a NIT as it's IMO a matter of personal preference. 

> I vaguely remember some issue with that.

BTW I looked it up and I was confusing it with the NS deletion issue Marc has found,  so no issue here

> This could be avoided when the node/resource is deleted and recreated again

Hmm is there something preventing us from using the same node for multiple tests ?

Reply:
>Hmm is there something preventing us from using the same node for multiple tests ?

Having new and independent resources doesn't connect unit tests that modify the resource (node). Simply recreating the node is easy, fast, and disconnect tests.